app/src/main/java/com/vero/tasky/auth/presentation/registration/RegistrationViewModel.kt

```diff
@@ -78,14 +79,12 @@ class RegistrationViewModel @Inject constructor(
     }
 
     private fun validatePassword() {
-        val result = PasswordValidator.validatePassword(
-            passwordValidationResult = registrationUseCases.validatePasswordUseCase(state.password)
-        )
+        val validationResult = registrationUseCases.validatePasswordUseCase(state.password)
+        val parsedResult = PasswordErrorParser.parse(validationResult)
 
-        if (result.isValid) {
-            signUp()
-        } else {
-            result.error?.let { showError(it) }
+        when(parsedResult) {
+            is PasswordParsedResult.Invalid -> showError(parsedResult.error)
+            PasswordParsedResult.Valid -> signUp()
         }
```

Comment:
This signUp() shouldnt be part of the validatePassword function tbh

The validatePassword, should only validate the password